[v17] Bump Node.js (v22) and Electron (v35), remove manual install of corepack from build scripts

ravicious opened the pull request:
Backport #53610.

Tag build: https://github.com/gravitational/teleport.e/actions/runs/14474495472

There were some conflicts in package.json since we don't backport dep updates, but nothing too bad. I had to manually force `@storybook/addon-actions` to be 8.3.4 since that's the version of Storybook we have on v17. Otherwise it was showing a warning about a missing peer dep (since we told pnpm to no longer install those manually).

What was surprising was that I didn't quite trust pnpm to resolve the conflicts. I was worried it's going to pull some deps from master into v17 or whatever. So instead of depending on `pnpm i` to resolve the conflits, I simply reverted all updates to pnpm-lock.yaml and re-run `pnpm i`. Indeed, after doing this the number of deps with unmet peer deps went from like 6 to just 1 (an unmet peer dep is not the same as missing peer dep; it just means that the version range is not compatible with the currently installed version).

dependabot[bot] committed:
Bump Node.js (v22) and Electron (v35), remove manual install of corepack from build scripts

* Bump the electron group with 3 updates

Bumps the electron group with 3 updates: [electron](https://github.com/electron/electron), [electron-builder](https://github.com/electron-userland/electron-builder/tree/HEAD/packages/electron-builder) and [electron-vite](https://github.com/alex8088/electron-vite).

Updates `electron` from 34.3.0 to 35.1.2
- [Release notes](https://github.com/electron/electron/releases)
- [Changelog](https://github.com/electron/electron/blob/main/docs/breaking-changes.md)
- [Commits](https://github.com/electron/electron/compare/v34.3.0...v35.1.2)

Updates `electron-builder` from 25.1.8 to 26.0.12
- [Release notes](https://github.com/electron-userland/electron-builder/releases)
- [Changelog](https://github.com/electron-userland/electron-builder/blob/master/packages/electron-builder/CHANGELOG.md)
- [Commits](https://github.com/electron-userland/electron-builder/commits/v26.0.12/packages/electron-builder)

Updates `electron-vite` from 3.0.0 to 3.1.0
- [Release notes](https://github.com/alex8088/electron-vite/releases)
- [Changelog](https://github.com/alex8088/electron-vite/blob/master/CHANGELOG.md)
- [Commits](https://github.com/alex8088/electron-vite/compare/v3.0.0...v3.1.0)

---
updated-dependencies:
- dependency-name: electron
  dependency-version: 35.1.2
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: electron
- dependency-name: electron-builder
  dependency-version: 26.0.12
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: electron
- dependency-name: electron-vite
  dependency-version: 3.1.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: electron
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump @electron/notarize from 2.5.0 to 3.0.1

Bumps [@electron/notarize](https://github.com/electron/notarize) from 2.5.0 to 3.0.1.
- [Release notes](https://github.com/electron/notarize/releases)
- [Changelog](https://github.com/electron/notarize/blob/main/.releaserc.json)
- [Commits](https://github.com/electron/notarize/compare/v2.5.0...v3.0.1)

---
updated-dependencies:
- dependency-name: "@electron/notarize"
  dependency-version: 3.0.1
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump node-pty from 1.1.0-beta14 to 1.1.0-beta33

Bumps [node-pty](https://github.com/microsoft/node-pty) from 1.1.0-beta14 to 1.1.0-beta33.
- [Release notes](https://github.com/microsoft/node-pty/releases)
- [Commits](https://github.com/microsoft/node-pty/commits/v1.1.0-beta33)

---
updated-dependencies:
- dependency-name: node-pty
  dependency-version: 1.1.0-beta33
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* Bump electron to 35.1.4

* Remove override for electron-vite

* Remove unnecessary node_gyp config option from .npmrc

It's enough to have a specific version of node-gyp in package.json and
that will cause node-pty to use that over the version shipped with pnpm.

* Bump Node.js version to 22.14.0

* Remove manual installation of corepack@0.31.0 and new integrity keys

* Fix electron-builder deps

This is done by removing electron-builder from package.json, running
pnpm install and then adding it back again and running pnpm install.

* Do not install missing peers to solve problem with electron-builder-squirrel-windows

* Move Windows signing configuration to `signtoolOptions`

* Update @types/node to v22

* Bring back node_gyp config option in .npmrc

This reverts commit b565095500eb7c8da7b64b54656f9e6a13526705.

Turns out it's actually needed, as on Windows pnpm uses 11.1.0 without
this.

* Update node-gyp to 11.2.0

* Update linux templates

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Rafał Cieślak <rafal.cieslak@goteleport.com>
Co-authored-by: Grzegorz Zdunek <grzegorz.zdunek@goteleport.com>

ravicious committed:
Force @storybook/addon-actions to 8.3.4

This removes a warning about an unmet peer dep:

.
└─┬ @storybook/addon-actions 8.6.7
  └── ✕ unmet peer storybook@^8.6.7: found 8.3.4

ravicious commented on pnpm-workspace.yaml:
```diff
@@ -8,3 +8,18 @@ onlyBuiltDependencies:
+# pull in deps that we don't use? Ultimately, the problem is that electron-builder-squirrel-windows
+# is defined as a peer dep of one of electron-builder deps, while in reality it's more like an
+# optional dep.
+autoInstallPeers: false
```
eslint-plugin-import is now missing, I'll have to look up what version we currently have on v17. It's weird though that `pnpm i --resolution-only` doesn't show any warning/error about this, I have to run `pnpm eslint` to see it.

gzdunek replied:
It's because `eslint-plugin-import` is an optional dependency https://github.com/import-js/eslint-import-resolver-typescript/blob/e36a0acebfd2846b58cc55c93db17e4d17b583cb/package.json#L73 💀 

I don't understand why, `eslint-import-resolver-typescript` needs `eslint-plugin-import` to work.

gzdunek committed:
Bump `node-abi` to 3.74.0

It matches the version on master.

gzdunek commented on web/packages/teleterm/package.json:
```diff
@@ -45,9 +45,9 @@
-    "electron-builder": "^25.1.8",
-    "electron-vite": "^3.0.0",
+    "electron": "35.1.4",
+    "electron-builder": "^26.0.12",
```
On master we install `node-abi@3.74.0`, I bumped it here.

ravicious committed:
Add eslint-plugin-import

ravicious committed:
Dedupe packages

ravicious committed:
Add override for nwsapi

As a result of deduping packages, nwsapi got updated to a version that
has a bug.

https://github.com/gravitational/teleport/pull/49613#discussion_r1867914741
https://github.com/gravitational/teleport/pull/52683#discussion_r1983842837

ravicious committed:
Merge branch 'branch/v17' into r7s/v17/backport-53610-node-v22

ravicious commented:
I verified that Connect and Web UI from the tag build work correctly.
